### PR TITLE
Revamping of Muon* algorithms for 2.3.12

### LINKED ISSUE #7: Helpers for Muon Efficiency SF tool

### PR DESCRIPTION
**Pull request by mmilesi**
This closes #7 and #8 , but opens #39  --> muon trigger efficiency SF and related systematics

### CODE REVIEW

**Comment by mmilesi** [comment]
Hi @gfacini , @kratsg ,

the code works fine on mc15, so it can be pushed into the new tag.

However, I have just sent an email to PAT help for clarification on the muon trigger SF part of the code (fuck, the documentation they made is true shit and does not really help understanding). In case they reply and I manage to fix also this, will it be possible for me to make a new commit on the muonEffSF branch and update this pull request?

Marco

**Comment by mmilesi** [comment]
My suggestion would be:

wait until tomorrow morning ( Sunday CERN time ), and if I haven't received feedback from PAT Help, let's merge what we have so far. I'll leave the trigger SF part commented.

@kratsg : can I still add commits to muonEffSF branch and update this pull req?
How can I do that?

**Comment by kratsg** [comment]
@mmilesi Just push new commits to the branch. The PR will automatically update. It's synced with the HEAD of that branch. If HEAD changes (new commits), the PR changes.

**Comment by mmilesi** [comment]
Hi @kratsg , do I have to pull master from origin and rebase as well after committing my changes on the branch?

**Comment by kratsg** [comment]
You only need to rebase changes if master was updated while you were editing. You've already rebased once, and there have not been any new pull requests merged to master, so you just need to `git push`. You would know if you need to rebase again when the pull request cannot be automatically merged.

File: util/test_multiAlgo.cxx [hunk 1]
```diff
@@ -164,7 +164,7 @@ int main( int argc, char* argv[] ) {
   //job.algsAdd( jetHistsAlgo_truth );
   //job.algsAdd( overlapRemoval );
   //job.algsAdd( jk_AntiKt10LC );
-  job.algsAdd( out_tree );
+  //job.algsAdd( out_tree );
```
Comment (kratsg) [inline util/test_multiAlgo.cxx]: Why comment it all out? Do they still not work?

**Comment by kratsg** [comment]
looks all good otherwise. Muons and Electrons definitely work.

**Pull request merged**